src/lib/case-studies/aveva-model.ts

```diff
@@ -0,0 +1,87 @@
ICAgIH1cblxuICAgIGlmICgnbWVtLWVuZXJneScgaW4gc3RhdGljUGFyYW1zKSB7XG4gICAgICB0aGlzLm1lbW9yeUVuZXJneSA9IHN0YXRpY1BhcmFtc1snbWVtLWVuZXJneSddIGFzIG51bWJlcjtcbiAgICB9XG5cbiAgICByZXR1cm4gdGhpcztcbiAgfVxuXG4gIC8qKlxuICAgKiBDYWxjdWxhdGUgdGhlIHRvdGFsIGVtaXNzaW9ucyBmb3IgYSBsaXN0IG9mIG9ic2VydmF0aW9uc1xuICAgKlxuICAgKiBFYWNoIE9ic2VydmF0aW9uIHJlcXVpcmU6XG4gICAqICBAcGFyYW0ge09iamVjdFtdfSBvYnNlcnZhdGlvbnNcbiAgICogIEBwYXJhbSB7c3RyaW5nfSBvYnNlcnZhdGlvbnNbXS50aW1lc3RhbXAgUkZDMzMzOSB0aW1lc3RhbXAgc3RyaW5nXG4gICAqICBAcGFyYW0ge251bWJlcn0gb2JzZXJ2YXRpb25zW10ubWVtLXV0aWwgcGVyY2VudGFnZSBtZW0gdXNhZ2VcbiAgICovXG4gIGFzeW5jIGNhbGN1bGF0ZShvYnNlcnZhdGlvbnM6IG9iamVjdCB8IG9iamVjdFtdIHwgdW5kZWZpbmVkKTogUHJvbWlzZTxhbnlbXT4ge1xuICAgIGlmIChvYnNlcnZhdGlvbnMgPT09IHVuZGVmaW5lZCkge1xuICAgICAgdGhyb3cgbmV3IEVycm9yKCdSZXF1aXJlZCBQYXJhbWV0ZXJzIG5vdCBwcm92aWRlZCcpO1xuICAgIH0gZWxzZSBpZiAoIUFycmF5LmlzQXJyYXkob2JzZXJ2YXRpb25zKSkge1xuICAgICAgdGhyb3cgbmV3IEVycm9yKCdPYnNlcnZhdGlvbnMgbXVzdCBiZSBhbiBhcnJheScpO1xuICAgIH1cbiAgICByZXR1cm4gb2JzZXJ2YXRpb25zLm1hcCgob2JzZXJ2YXRpb246IEtleVZhbHVlUGFpcikgPT4ge1xuICAgICAgdGhpcy5jb25maWd1cmUodGhpcy5uYW1lISwgb2JzZXJ2YXRpb24pO1xuICAgICAgb2JzZXJ2YXRpb25bJ2UtbWVtJ10gPSB0aGlzLmNhbGN1bGF0ZUVuZXJneShvYnNlcnZhdGlvbik7XG4gICAgICByZXR1cm4gb2JzZXJ2YXRpb247XG4gICAgfSk7XG4gIH1cblxuICAvKipcbiAgICogUmV0dXJucyBtb2RlbCBpZGVudGlmaWVyXG4gICAqL1xuICBtb2RlbElkZW50aWZpZXIoKTogc3RyaW5nIHtcbiAgICByZXR1cm4gJ2UtbWVtJztcbiAgfVxuXG4gIC8qKlxuICAgKiBDYWxjdWxhdGVzIHRoZSBlbmVyZ3kgY29uc3VtcHRpb24gZm9yIGEgc2luZ2xlIG9ic2VydmF0aW9uXG4gICAqIHJlcXVpcmVzXG4gICAqXG4gICAqIG1lbS11dGlsOiByYW0gdXNhZ2UgaW4gcGVyY2VudGFnZVxuICAgKiB0aW1lc3RhbXA6IFJGQzMzMzkgdGltZXN0YW1wIHN0cmluZ1xuICAgKlxuICAgKiBtdWx0aXBsaWVzIG1lbW9yeSB1c2VkIChHQikgYnkgYSBjb2VmZmljaWVudCAod2gvR0IpIGFuZCBjb252ZXJ0cyB0byBrd2hcbiAgICovXG4gIHByaXZhdGUgY2FsY3VsYXRlRW5lcmd5KG9ic2VydmF0aW9uOiBLZXlWYWx1ZVBhaXIpIHtcbiAgICBpZiAoISgnbWVtLXV0aWwnIGluIG9ic2VydmF0aW9uKSB8fCAhKCd0aW1lc3RhbXAnIGluIG9ic2VydmF0aW9uKSkge1xuICAgICAgdGhyb3cgbmV3IEVycm9yKFxuICAgICAgICAnUmVxdWlyZWQgUGFyYW1ldGVycyBkdXJhdGlvbixjcHUtdXRpbCx0aW1lc3RhbXAgbm90IHByb3ZpZGVkIGZvciBvYnNlcnZhdGlvbidcbiAgICAgICk7XG4gICAgfVxuICAgIGlmICh0aGlzLm1lbW9yeUFsbG9jYXRpb24gPT09IDApIHtcbiAgICAgIHRocm93IG5ldyBFcnJvcihcbiAgICAgICAgJ1JlcXVpcmVkIFBhcmFtZXRlcjogbWVtLWFsbG9jIG5vdCBwcm92aWRlZCBpbiBjb25maWd1cmUnXG4gICAgICApO1xuICAgIH1cbiAgICBpZiAodGhpcy5tZW1vcnlFbmVyZ3kgPT09IDApIHtcbiAgICAgIHRocm93IG5ldyBFcnJvcihcbiAgICAgICAgJ1JlcXVpcmVkIFBhcmFtZXRlcjogbWVtLWVuZXJneSBub3QgcHJvdmlkZWQgaW4gY29uZmlndXJlJ1xuICAgICAgKTtcbiAgICB9XG5cbiAgICBjb25zdCBtZW1fYWxsb2MgPSB0aGlzLm1lbW9yeUFsbG9jYXRpb247XG4gICAgLy8gICAgY29udmVydCBjcHUgdXNhZ2UgdG8gcGVyY2VudGFnZVxuICAgIGNvbnN0IG1lbV91dGlsID0gb2JzZXJ2YXRpb25bJ21lbS11dGlsJ107XG4gICAgaWYgKG1lbV91dGlsIDwgMCB8fCBtZW1fdXRpbCA+IDEwMCkge1xuICAgICAgdGhyb3cgbmV3IEVycm9yKCdjcHUgdXNhZ2UgbXVzdCBiZSBiZXR3ZWVuIDAgYW5kIDEwMCcpO1xuICAgIH1cblxuICAgIGNvbnN0IG1lbV9lbmVyZ3kgPSB0aGlzLm1lbW9yeUVuZXJneTtcblxuICAgIHJldHVybiAobWVtX2FsbG9jICogKG1lbV91dGlsIC8gMTAwKSAqIG1lbV9lbmVyZ3kpIC8gMTAwMDtcbiAgfVxufVxuIl19
```

Comment:
what is this?

Reply:
urgh - an accidental paste from the js file

Reply:
fixing in new pr